packages/ra-core/src/controller/useList.ts

```diff
@@ -0,0 +1,233 @@
+import { useCallback, useEffect } from 'react';
+// @ts-ignore
+import { useSelector, useDispatch } from 'react-redux';
+import { parse, stringify } from 'query-string';
+import { push } from 'connected-react-router';
+import pickBy from 'lodash/pickBy';
+
+import queryReducer, {
+    SET_SORT,
+    SET_PAGE,
+    SET_PER_PAGE,
+    SORT_ASC,
+} from '../reducer/admin/resource/list/queryReducer';
+import { crudGetList } from '../actions/dataActions';
+import { changeListParams, ListParams } from '../actions/listActions';
+import { Sort, ReduxState, Identifier, RecordMap } from '../types';
+import { Location } from 'history';
+
+interface Props {
+    filter?: object;
+    filterDefaultValues?: object;
+    perPage?: number;
+    sort?: Sort;
+    location: Location;
+    resource: string;
+}
+
+interface Query extends ListParams {
+    data: RecordMap;
+    ids: Identifier[];
+    total: number;
+    requestSignature: any[];
+}
+
+interface Actions {
+    changeParams: (action: any) => void;
+    setPage: (page: number) => void;
+    setPerPage: (pageSize: number) => void;
+    setSort: (sort: Sort) => void;
+}
+
+const useList = ({
```

Comment:
I'd call it useListParams

Reply:
`useCrudList` ?

Reply:
we tend to use the Crud word for communication with the dataProvider. this hook does not communicate with the dataProvider, so I'm -1 for that name.